src/merkle/merkle_tree.rs

```diff
@@ -120,7 +117,7 @@ impl MerkleTree {
     /// Returns an iterator over the leaves of this [MerkleTree].
     pub fn leaves(&self) -> impl Iterator<Item = (u64, &Word)> {
         let leaves_start = self.nodes.len() / 2;
-        self.nodes.iter().skip(leaves_start).enumerate().map(|(i, v)| (i as u64, v))
+        self.nodes.iter().skip(leaves_start).enumerate().map(|(i, v)| (i as u64, &**v))
```

Comment:
nit: instead of doing `&**v`, you should be able to do something like `v.deref()`. I think it would be a bit cleaner.